code/game/objects/items/devices/atomic_disassembler.dm

```diff
@@ -0,0 +1,156 @@
+		user << "<span class='notice'>You place \the [W] into its slot and connect it to \the [src]'s feeding hose.</span>"
+		src.canister = W
+		user.drop_item()
+		W.loc = src
+		return
+	if(!power_cell)
+		user << "<span class='warning'>\The [src] requires a power cell.</span>"
+		return
+	if(power_cell.charge <= 0)
+		user << "<span class='warning'>\The [src]'s power cell is out of charge.</span>"
+		return
+	if(!canister)
+		user << "<span class='warning'>\The [src] will not function without a compressed matter canister.</span>"
+		return
+	if(canister.stored_matter == canister.matter_cap)
+		user << "<span class='warning'>\The [src]'s matter canister is full - replace it.</span>"
+		return
+	if(istype(W, /obj/item/device/atomic_disassembler))
+		var/boom = alert(user, "This probably isn't wise...", "Ignore the warning labels?", "Yes", "No")
+
+		if(boom == "No" || !user.canUseTopic(src) || !W)
+			return
+
+		user.visible_message("<span class='warning'>[user] feeds \the [W] into \the [src], and a screech erupts as they catastrophically malfunction!</span>", \
+							 "<span class='boldannounce'>You feed \the [W] into \the [src]. <i>That wasn't a good idea.</i></span>")
+		playsound(get_turf(src), 'sound/effects/supermatter.ogg', 100, 1)
+		src.icon_state = "atom_emagged"
+		qdel(W)
+		sleep(20)
+		explosion(src.loc, -1, 2, 4, flame_range = 2)
+		qdel(src)
+		return
+	if((is_type_in_list(W, blacklisted_items) && !emagged) || is_type_in_list(W, really_blacklisted_items))
+		user << "<span class='warning'>\The [src]'s atomization field does not accept \the [W].</span>"
+		return
+	else
+		user.visible_message("<span class='notice'>[user] feeds \the [W] into \the [src]'s atomization field.</span>", \
+							 "<span class='info'>\The [W] has been broken down.</span>")
+		canister.update_matter(10 * W.w_class)
+		power_cell.charge -= 10 * W.w_class
+		if(power_cell.charge < 0)
+			power_cell.charge = 0
+		power_cell.update_icon()
+		playsound(src, 'sound/effects/EMPulse.ogg', 25, 1)
+		user.drop_item()
+		qdel(W)
+
+/obj/item/device/atomic_disassembler/verb/eject_power_cell()
+	set name = "Eject Power Cell"
+	set category = "Atomic Disassembler"
+	set src in usr
+
+	if(usr.canUseTopic(src))
+		if(!power_cell)
+			usr << "<span class='warning'>\The [src] doesn't have a power cell installed.</span>"
+			return 0
+		usr << "<span class='notice'>You eject \the [src]'s power cell.</span>"
+		icon_state = "atom_inactive"
+		src.power_cell.loc = usr.loc
+		src.power_cell = null
+
+/obj/item/device/atomic_disassembler/verb/eject_canister()
+	set name = "Eject Matter Canister"
+	set category = "Atomic Disassembler"
+	set src in usr
+
+	if(usr.canUseTopic(src))
+		if(!canister)
+			usr << "<span class='warning'>\The [src] has no compressed matter canister hooked up.</span>"
+			return 0
+		usr << "<span class='notice'>You unhook the compressed matter canister from \the [src].</span>"
+		src.canister.loc = usr.loc
+		src.canister = null
```

Comment:
1) Do we really want more verbs like this or should it be an action button or what? I'm asking because I actually don't know what the ideal way to handle this kind of case is.
2) Should probably use put_in_hands() so the canister doesn't fall onto the floor